### ISSUE TITLE
A Problem in fgas.cpp

### ISSUE THREAD

**Issue by IffyZh** [abmantz/fgas-cosmo#1]
There is a return value type error at fgas.cpp line 102 and my compiler failed to compile it due to that reason. I use Visual Studio 2022 to run it though I do not think it is an error about compiler. The type of value "fin" is std::ifstream yet the return value of "Dataset::init" is bool (at line 86 and 87). 

I wonder is it a mistake there or it is special usage? If it is not a mistake, how could I do to compile it? I really appreciate for help. 
![20240423150215](https://github.com/abmantz/fgas-cosmo/assets/167850075/1f986d59-26bb-46ba-923d-3284092f4b07)

**Comment by abmantz** [comment]
Yeah, looks like that should be `return true;`. Less picky/older compilers probably treated `fin` as `true` unless it was zero or a null pointer or such. At first glance that change should do it, but let me know whether or not it works.
